### PR TITLE
Nameservice module cli, grpc test cases

### PR DESCRIPTION
**Pull request by gsk967**
Closes: #https://github.com/vulcanize/dxns/issues/7

## Description
This pull request contains the test cases for nameservice, bond module 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/nameservice/keeper/grpc_query_test.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+	testCases := []struct {
+		msg string
+		req *nameservicetypes.QueryParamsRequest
+	}{
+		{
+			"Get Params",
+			&nameservicetypes.QueryParamsRequest{},
+		},
+	}
+	for _, test := range testCases {
```
Comment (anilCSE) [inline x/nameservice/keeper/grpc_query_test.go:23]: Let's avoid tabular tests if we have just one test case

File: x/nameservice/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+			resp, _ := grpcClient.Params(context.Background(), test.req)
+			defaultParams := nameservicetypes.DefaultParams()
+			suite.Require().Equal(defaultParams.String(), resp.GetParams().String())
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestGrpcGetRecordLists() {
+	grpcClient, ctx := suite.queryClient, suite.ctx
+	sr := suite.Require()
```
Comment (anilCSE) [inline x/nameservice/keeper/grpc_query_test.go:34]: Hmm this is creating some readability issues. Let's use it directly?

Comment (gsk967) [reply x/nameservice/keeper/grpc_query_test.go:34]: Okay

File: x/nameservice/keeper/grpc_query_test.go (hunk 1 above)
Comment (gsk967) [reply x/nameservice/keeper/grpc_query_test.go:23]: Okay

File: x/nameservice/client/testutil/tx.go [hunk 3]
```diff
@@ -0,0 +1,910 @@
...
+				fmt.Sprintf("--%s=json", tmcli.OutputFlag),
+				fmt.Sprintf("--%s=%s", flags.FlagBroadcastMode, flags.BroadcastBlock),
+				fmt.Sprintf("--%s=%s", flags.FlagFees, fmt.Sprintf("3%s", s.cfg.BondDenom)),
+			},
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(fmt.Sprintf("Case %s", tc.name), func() {
```
Comment (bipulprasad) [inline x/nameservice/client/testutil/tx.go:121]: Test cases for `NewCreateBondCmd` already exist in the bond module cli testutil. No need to run test cases here.

Simply create a bond and assert that there is no error in the process.

File: x/nameservice/client/testutil/tx.go [hunk 4]
```diff
@@ -0,0 +1,910 @@
...
+				sr.NoError(err)
+				sr.Zero(d.Code)
+				// post-run
+				tc.postRun(recordId, s)
+			}
+		})
+	}
+}
+
+//
```
Comment (bipulprasad) [inline x/nameservice/client/testutil/tx.go:658]: Remove all these lines of commented code, if not necessary.

File: x/nameservice/keeper/grpc_query_test.go [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+				if test.createRecord {
+					recordId = resp.GetRecords()[0].GetId()
+					sr.NotZero(resp.GetRecords())
+					sr.Equal(resp.GetRecords()[0].GetBondId(), suite.bond.GetId())
+				}
+			}
+		})
+	}
+
+	// Get the records by record id
```
Comment (bipulprasad) [inline x/nameservice/keeper/grpc_query_test.go:87]: Maybe add a log for these different record fetching scenarios, similar to what you did in `client/testutil/query.go`.

**Pull request merged**